src/project-manager.ts

```diff
@@ -680,13 +891,39 @@ export class ProjectConfiguration {
 	// (https://github.com/Microsoft/TypeScript-wiki/blob/master/Architectural-Overview.md#data-structures)
 	private program?: ts.Program;
 
+	/**
+	 * Object TS service will use to fetch content of source files
+	 */
 	private host?: InMemoryLanguageServiceHost;
 
+	/**
+	 * Local file cache
+	 */
 	private fs: InMemoryFileSystem;
+
+	/**
+	 * Path to configuration file (tsconfig.json/jsconfig.json)
+	 */
 	private configFileName: string; // path to [tj]sconfig.json, if exists
+
+	/**
+	 * Configuration JSON object. May be used when there is no real configuration file to parse and use
+	 */
 	private configContent: any;
+
+	/**
+	 * Tracks source file -> version
+	 */
 	private versions: Map<string, number>;
+
+	/**
+	 * Enables module resolution tracing (done by TS service)
+	 */
 	private traceModuleResolution: boolean;
+
+	/**
+	 * Root folder
```

Comment:
path, relative to workspace hierarchy. Updated documentation and renamed field/parameter